### ISSUE TITLE
seq repository not found

### ISSUE THREAD

**Issue by dieggsy** [raxod502/straight.el#170]
Recently installed a different OS, upon setting up emacs I got the following error on building the flycheck package:

```
Warning (initialization): An error occurred while loading ‘/home/diego/.emacs.d/init.el’:

error: Command failed: git clone --origin origin --no-checkout https://github.com/emacsmirror/seq.git seq (output: "Cloning into 'seq'...
remote: Repository not found.
fatal: repository 'https://github.com/emacsmirror/seq.git/' not found
") (default-directory: "/home/diego/.emacs.d/straight/repos/")

To ensure normal operation, you should investigate and remove the
cause of the error in your initialization file.  Start Emacs with
the ‘--debug-init’ option to view a complete error backtrace.
```

**Comment by raxod502** [comment]
See https://github.com/emacsmirror/epkgs/issues/2.

**Comment by raxod502** [comment]
Unfortunately this development means that I will have to resurrect support for GNU ELPA, which was dropped in https://github.com/raxod502/straight.el/commit/2721f10370bd19ecfb00f05da324e556759518d9.

**Comment by raxod502** [comment]
In the meantime, you can do:

```elisp
(straight-register-package
  '(seq :repo "https://git.savannah.gnu.org/git/emacs/elpa.git" :files ("packages/seq/*.el")))
```

**Comment by SkySkimmer** [comment]
Is there some reason not to use the builtin seq? I've overridden recipe `(seq :type git :host github :repo "raxod502/straight.el")` which is a really stupid hack but seems to work. A saner way to do it would be nice of course.

**Comment by raxod502** [comment]
`straight.el` [does use the built-in `seq`](https://github.com/raxod502/straight.el/blob/3aada435981e2ba0063f12ce0c4d7927a6bd42d7/straight.el#L1690). I'm not sure why you would have to use that hack, and am also unsure of what it would even do.

**Comment by SkySkimmer** [comment]
It's because I was in a broken state with emacsmirror still believing in seq.

**Comment by raxod502** [comment]
`straight.el` should not even attempt to look up a recipe for a package if it is built in. (Yes `M-x straight-get-recipe` suggests to the contrary, but if you look at the code for `straight--convert-recipe` you'll see that it doesn't get to `straight-get-recipe` in the case that `package-built-in-p` returns true.)

**Comment by SkySkimmer** [comment]
AFAICT in `convert-recipe` the call to `recipes-retrieve` comes before `package-built-in-p` and if the former succeeds the later is not executed (along with requiring package).

**Comment by raxod502** [comment]
Oh, you are right. I misremembered how this logic worked. Currently, it's set up so that packages are installed from a remote source if possible, since the Emacs source is most likely to be out of date. The only thing `straight.el` uses `package-built-in-p` for is deciding whether or not to signal an error when a package is unavailable.

Sorry about the confusion. And now I see what you are trying to do. I think that the best way to handle this (interface-wise) would be via

```
(straight-register-package '(seq :local-repo nil))
```

although I'm not sure this works at the moment. This could be wrapped in a helper function to make the call more intuitive (like `straight-use-package-lazy`):

```
(straight-use-builtin-package 'seq)
```

I'll split this issue into:

* adding such functionality, and
* resurrecting GNU ELPA support so that an up-to-date version of `seq.el` can be used instead of Emacs'.

**Comment by SkySkimmer** [comment]
Sounds good, except I think of it more as having a `:type 'built-in` (with most vc operations being noops, local-repo-name returning nil).

**Comment by raxod502** [comment]
Ah! That makes much more sense, indeed.

Although the VC abstraction still kind of assumes that there is a local repository currently, so some rethinking there will be necessary.

I've split this issue into #171 (which will fix `seq.el` not being available) and #172 (which would allow usage of the built-in `seq.el` even when the local recipe repository tells `straight.el` to fetch it from EmacsMirror).

**Comment by dieggsy** [comment]
@raxod502 the `straight-register-package` workaround did the trick. thanks for the prompt and thorough replies as always.
